### PR TITLE
Remove native id check in pda creation

### PR DESCRIPTION
**Pull request by jackcmay**
#### Problem

The native program id checks when creating program-derived addresses is unnecessary and special cases native ids over all other program ids for no known reason.

#### Summary of Changes

Remove it

Fixes #

### CODE REVIEW

File: programs/bpf/rust/invoke/src/lib.rs [hunk 1]
```diff
@@ -263,10 +263,9 @@ fn process_instruction(
                     )?,
                     accounts[DERIVED_KEY1_INDEX].key
                 );
-                let not_native_program_id = Pubkey::new_from_array([6u8; 32]);
-                assert!(!not_native_program_id.is_native_program_id());
+                let new_program_id = Pubkey::new_from_array([6u8; 32]);
```
Comment (t-nelson) [inline programs/bpf/rust/invoke/src/lib.rs:266]: nit: Could switch these to `Pubkey::new_unique()` while we're here

File: programs/bpf_loader/src/syscalls.rs [hunk 2]
```diff
@@ -253,6 +253,7 @@ pub fn bind_syscall_context_objects<'a>(
             compute_meter: invoke_context.get_compute_meter(),
             loader_id,
             enforce_aligned_host_addrs,
+            allow_native_ids: invoke_context.is_feature_active(&allow_native_ids::id()),
```
Comment (t-nelson) [inline programs/bpf_loader/src/syscalls.rs:256]: nit: could do this once and reuse

File: programs/bpf_loader/src/syscalls.rs [hunk 3]
```diff
@@ -834,12 +836,42 @@ fn translate_program_address_inputs<'a>(
     Ok((seeds, program_id))
 }
 
+fn is_native_id(seeds: &[&[u8]], program_id: &Pubkey) -> bool {
+    use solana_sdk::{config, feature, secp256k1_program, stake, system_program, vote};
+    // Does more than just check native ids in order to emulate the same failure
+    // signature that `compute_program_address` had before the rmemal of the
```
Comment (t-nelson) [inline programs/bpf_loader/src/syscalls.rs:842]:
```suggestion
    // signature that `compute_program_address` had before the removal of the
```
right?

**Review by t-nelson — APPROVED** [review]
Looks great!  Just a couple nits

File: programs/bpf/rust/invoke/src/lib.rs (hunk 1 above)
Comment (jackcmay) [reply programs/bpf/rust/invoke/src/lib.rs:266]: Yeah, I'd like to keep this pr clean, can do a mas s// to new_unique in a separate pr

**Pull request merged**